### PR TITLE
first cut at go mod

### PR DESCRIPTION
**Pull request by vaikas**

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -21,19 +21,24 @@ FLOATING_DEPS=(
 )
```
Comment (mattmoor) [inline hack/update-deps.sh:21]: I think you need to change above to have VERSION and @VERSION?

Comment (vaikas) [reply hack/update-deps.sh:28]: yes, also had forgotten the gomod line. still working...

File: hack/update-deps.sh [hunk 2]
```diff
@@ -12,28 +12,40 @@ set -o pipefail
...
+
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/pkg"
-  "knative.dev/eventing"
-  "knative.dev/test-infra"
+  "knative.dev/pkg@{$VERSION}"
+  "knative.dev/eventing@{$VERSION}"
+  "knative.dev/test-infra@{$VERSION}"
```
Comment (mattmoor) [inline hack/update-deps.sh:27]: These are incorrect, the `$` goes outside of the curly braces.  It broke in sample-controller this morning.

File: hack/update-deps.sh [hunk 3]
```diff
@@ -12,28 +12,40 @@ set -o pipefail
 
 cd ${ROOT_DIR}
 
+# This controls the release branch we track.
+VERSION="master"
+
+# We need these flags for things to work properly.
+export GO111MODULE=on
+export GOFLAGS=-mod=vendor
```
Comment (mattmoor) [inline hack/update-deps.sh:20]: Hmm, we aren't running the license code, so we don't need this yet.  Having it here will break `--upgrade`

File: test/upload-test-images.sh [hunk 4]
```diff
@@ -5,8 +5,11 @@
 
 : ${KO_DOCKER_REPO:?"You must set 'KO_DOCKER_REPO', see DEVELOPMENT.md"}
 
-cat | ko resolve --strict -Bf - <<EOF
+export GO111MODULE=on
+export GOFLAGS=-mod=vendor
+
+cat | ko resolve --strict -RBf - <<EOF
```
Comment (mattmoor) [inline test/upload-test-images.sh:11]: R is meaningless here

File: .github/workflows/kind-e2e.yaml [hunk 5]
```diff
@@ -29,8 +29,10 @@ jobs:
         echo "::add-path::$(go env GOPATH)/bin"
 
     - name: Install Dependencies
+      # Note the @master until the release version supports building images
+      # properly from the vendor directories
       run: |
-        GO111MODULE=on go get github.com/google/ko/cmd/ko
+        GO111MODULE=on go get github.com/google/ko/cmd/ko@master
```
Comment (mattmoor) [inline .github/workflows/kind-e2e.yaml:35]: master is now released, so you should be good to drop this.

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,72 @@
+module github.com/vmware-tanzu/sources-for-knative
+
+go 1.13
```
Comment (mattmoor) [inline go.mod:3]: 1.14

File: config/200-clusterrole.yaml [hunk 7]
```diff
@@ -24,6 +24,9 @@ rules:
   - apiGroups: [""]
     resources: ["configmaps", "services", "secrets", "events", "serviceaccounts"]
     verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
+  - apiGroups: [""]
+    resources: ["namespaces"]
```
Comment (mattmoor) [inline config/200-clusterrole.yaml:28]: you can just add this to the entry above 🤷

Comment (vaikas) [reply config/200-clusterrole.yaml:28]: don't want to create/delete rights.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**